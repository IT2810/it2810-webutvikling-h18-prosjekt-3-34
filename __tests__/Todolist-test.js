import React from "react";
import Todolist from ".././components/todolist";
import Itemcomponent from ".././components/itemcomponent";
import { ScrollView } from "react-native";
import ShallowRenderer from "react-test-renderer/shallow";

let items = [
  { id: 1, inputid: "input1", type: "todo", text: "kjop melk", done: true }
];

//Snapshot test

const shallow = new ShallowRenderer();
shallow.render(<Todolist items={items} />);
const result = shallow.getRenderOutput();

test("renders correctly", () => {
  expect(result).toMatchSnapshot();
  expect(result.type).toBe(ScrollView);
});

test("renders child correctly", () => {
  expect(result.props.children).toEqual([
    <Itemcomponent key="input1" id={1} text="kjop melk" status={true} />
  ]);
});
