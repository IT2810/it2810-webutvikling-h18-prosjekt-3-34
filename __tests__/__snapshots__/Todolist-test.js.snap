// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ScrollViewMock
  style={
    Object {
      "borderRadius": 4,
      "borderWidth": 1,
      "marginTop": 5,
      "maxHeight": "38%",
      "paddingLeft": 10,
      "paddingRight": 10,
      "width": "85%",
    }
  }
>
  <ItemComponent
    id={1}
    status={true}
    text="kjop melk"
  />
</ScrollViewMock>
`;
